lockfile/src/lib.rs

```diff
@@ -440,4 +457,55 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn skip_setup_with_pyproject() {
+        // Create desired directory structure.
+        let tempdir = tempfile::tempdir().unwrap();
+        let files = [
+            tempdir.path().join("pyproject.toml"),
+            tempdir.path().join("setup.py"),
+            tempdir.path().join("a/setup.py"),
+        ];
+        for file in &files {
+            let dir = file.parent().unwrap();
+            fs::create_dir_all(dir).unwrap();
+            File::create(file).unwrap();
+        }
+
+        // Find manifest files.
+        let lockable_files = find_lockable_files_at(tempdir.path());
+
+        // Compare results.
+        let expected =
+            vec![(tempdir.path().join("pyproject.toml").to_path_buf(), LockfileFormat::Pip)];
+        assert_eq!(lockable_files, expected);
+    }
+
+    #[test]
+    fn setup_without_pyproject() {
+        // Create desired directory structure.
+        let tempdir = tempfile::tempdir().unwrap();
+        let files = [
+            tempdir.path().join("setup.py"),
+            tempdir.path().join("a/setup.py"),
+            tempdir.path().join("b/pyproject.toml"),
+        ];
+        for file in &files {
+            let dir = file.parent().unwrap();
+            fs::create_dir_all(dir).unwrap();
+            File::create(file).unwrap();
+        }
+
+        // Find manifest files.
+        let lockable_files = find_lockable_files_at(tempdir.path());
+
+        // Compare results.
+        let expected = vec![
+            (tempdir.path().join("b/pyproject.toml").to_path_buf(), LockfileFormat::Pip),
+            (tempdir.path().join("a/setup.py").to_path_buf(), LockfileFormat::Pip),
+            (tempdir.path().join("setup.py").to_path_buf(), LockfileFormat::Pip),
+        ];
+        assert_eq!(lockable_files, expected);
```

Comment:
It seems less than ideal that the test requires a specific ordering of the results.

Maybe we should sort `lockable_files` before comparing it against `expected` (We would also need to change the order of `expected` to match)?

```rust
lockable_files.sort_by(|(a, _), (b, _)| a.cmp(b));
```

In this case, you could even sort by length if you want to keep the `expected` list pretty:

```rust
// Sort the list, longest path first
lockable_files.sort_by_key(|(p, _)| usize::MAX - p.as_os_str().len());
```

Reply:
Looks like this issue is actually causing the tests to fail...